packages/shylu/shylu_node/hts/src/shylu_hts_impl_def.hpp

```diff
@@ -4119,17 +4129,14 @@ RecursiveTri::solve (const Sclr* b, Sclr* x, const Int ldx,
     if ( ! s.empty() && (s.parallel() || tid == 0)) {
       rbwait(s_done, s_ids, s_idx, i, done_symbol);
       s.n1Axpy(x_osi, ldx, nrhs, x_os, ldx, tid);
-      s_done[rb_p2p_sub2ind(i, tid)] = done_symbol;
 #ifdef _OPENMP
 #     pragma omp flush
 #endif
+      s_done[rb_p2p_sub2ind(i, tid)] = done_symbol;
```

Comment:
This store not getting flushed to global memory in some processes?

Reply:
The `volatile` keyword leads to an implied flush of that variable. It's equivalent to the `list` clause in the OpenMP 5.0 `flush` construct, with the volatile variables in the list.